client/actions/assignments/ui.js

```diff
@@ -188,6 +188,24 @@ const reassign = checkPermission(
 )
 
 /**
+ * Action for opening modal to edit assignment's priority
+ *
+ */
+const _openEditPriorityModal = (assignment) => (
+    (dispatch) => dispatch(_openActionModal(
+        assignment,
+        ASSIGNMENTS.ITEM_ACTIONS.EDIT_PRIORITY.label,
+        'edit_priority'
+    ))
+)
+
+const editPriority = checkPermission(
```

Comment:
Maybe we should not use the `checkPermission` functionality but instead not show the action if there is no permission.
This removes the need for these extra dispatches.

Reply:
I have PR #509 where I cover privileges for actions. So, I'll take this up in that as I inject privileges as prop in that PR already.